Kernel/Bus/PCI/Device.cpp

```diff
@@ -63,4 +68,104 @@ void Device::disable_extended_message_signalled_interrupts()
     }
 }
 
+PCI::InterruptType Device::get_interrupt_type()
+{
+    return m_interrupts.m_type;
+}
+
+// Reserve numbers_of_irqs for this device.
+// Returns the interrupt type that was reserved. It is a noop
+// for pin based interrupts as there is nothing left to do
+// The second parameter msi is used by the driver to indicate its
+// intent to use message signalled interrupts. But MSI(x) is preferred
+// over MSI
```

Comment:
Comments in this file are missing periods everywhere. (GitHub's file comment UI is currently broken for me, so I can't use that)